train.py

```diff
@@ -94,34 +118,41 @@
 
             # Training
             num_batches = X_train.shape[0] // BATCH_SIZE
-            for b in range(num_batches):
+            for b in tqdm(range(num_batches)):
                 x_batch, y_batch = next(train_batch_generator)
                 seq_len = np.array([list(x).index(0) + 1 for x in x_batch])  # actual lengths of sequences
-                loss_tr, acc, _ = sess.run([loss, accuracy, optimizer],
+                loss_tr, acc, _ ,summary= sess.run([loss, accuracy, optimizer,merged],
                                            feed_dict={batch_ph: x_batch,
                                                       target_ph: y_batch,
                                                       seq_len_ph: seq_len,
                                                       keep_prob_ph: KEEP_PROB})
                 accuracy_train += acc
                 loss_train = loss_tr * DELTA + loss_train * (1 - DELTA)
+                train_writer.add_summary(summary, b)
             accuracy_train /= num_batches
 
             # Testing
-            num_batches = X_test.shape[0] // BATCH_SIZE
-            for b in range(num_batches):
-                x_batch, y_batch = next(test_batch_generator)
-                seq_len = np.array([list(x).index(0) + 1 for x in x_batch])  # actual lengths of sequences
-                loss_test_batch, acc = sess.run([loss, accuracy],
-                                                feed_dict={batch_ph: x_batch,
-                                                           target_ph: y_batch,
-                                                           seq_len_ph: seq_len,
-                                                           keep_prob_ph: 1.0})
-                accuracy_test += acc
-                loss_test += loss_test_batch
-            accuracy_test /= num_batches
-            loss_test /= num_batches
-
-            print("loss: {:.3f}, val_loss: {:.3f}, acc: {:.3f}, val_acc: {:.3f}".format(
-                loss_train, loss_test, accuracy_train, accuracy_test
-            ))
-        saver.save(sess, "model")
+            if epoch%10 == 0:
```

Comment:
There are just 3 epochs now. So this condition makes no sense. If test subset is small, we can perform evaluation each epoch, there is no problem.

Reply:
Yes, sorry for this line, I was running on my very slowly CPU notebook and wanted to speed up the progress, It was too late after I ran over all the code and find out there are only 3 epochs.